src/include/snapper/dialogs.rb

```diff
@@ -1233,6 +1018,196 @@ def ShowDialog
+      enable_buttons([:modify, :show, :delete], selected.size == 1)
+      enable_buttons([:delete], selected.size >= 1)
+
+    end
+
+    def get_snapshot_items
+      snapshot_items = []
+
+      Snapper.snapshots.each_with_index do |s,i|
+        num = s["num"] || 0
+        start_date = (num != 0) ? timestring(s["date"]) : ""
+        end_date = ""
+        userdata = Snapper.userdata_to_string(s["userdata"])
+        desc = s["description"].to_s
+        if s["type"] == :SINGLE
+          type = _("Single")
+        elsif s["type"] == :POST
+          pre = s["pre_num"] || 0 # pre canot be 0
+          index = Ops.get(Snapper.id2index, pre, -1)
+          if pre == 0 || index == -1
+            Builtins.y2warning(
+              "something wrong - pre:%1, index:%2",
+              pre,
+              index
+            )
+            next
+          end
+          desc = Ops.get_string(Snapper.snapshots, [index, "description"], "")
+          end_date = start_date
+          start_date = timestring(Snapper.snapshots[index]["date"])
+          num = "%{pre} & %{post}" % { :pre => pre, :post => num }
+          type = _("Pre & Post")
+        else
+          # 0 means there's no post
+          if s["post_num"].to_i == 0
+            Builtins.y2milestone("pre snappshot %1 does not have post", num)
+            type = _("Pre")
+          else
+            Builtins.y2milestone("skipping pre snapshot: %1", num)
+            next
+          end
+        end
+        snapshot_items << Item(Id(i), num, type, start_date, end_date, desc, userdata)
+      end
+      snapshot_items
+    end
+
+    def config_select
+      HBox(
+        # combo box label
+        Label(_("Current Configuration")),
+        ComboBox(Id(:configs), Opt(:notify), "", Builtins.maplist(Snapper.configs) do |config|
+          Item(Id(config), config, config == Snapper.current_config)
+        end),
+        HStretch()
+      )
+    end
+
+    def snapshots_table
+      VBox(
+        config_select,
+        Table(
+          Id(:snapshots_table),
+          Opt(:notify, :keepSorting, :multiSelection, :immediate),
+          Header(
+            # table header
+            _("ID"),
+            _("Type"),
+            _("Start Date"),
+            _("End Date"),
+            _("Description"),
+            _("User Data")
+          ),
+          get_snapshot_items
+        ),
+        snapshots_table_footer
+      )
+    end
+
+    def snapshots_table_footer
+      HBox(
+        # button label
+        PushButton(Id(:show), Opt(:default), _("Show Changes")),
+        PushButton(Id(:create), Label.CreateButton),
+        # button label
+        PushButton(Id(:modify), _("Modify")),
+        PushButton(Id(:delete), Label.DeleteButton),
+        HStretch()
+      )
+    end
+
+    def selected_snapshots(selection)
+      selection.map do |s|
+        Snapper.snapshots[s]
+      end
+    end
+
+    def pre_lonely_snapshots
```

Comment:
this name is bit mixed and do not reflect that it do not return snapshots, only its num, so maybe better `lonely_pre_snapshots_num`